src/core.jl

```diff
@@ -1230,6 +1230,19 @@ Throws a `NodeNameNotFound` exception if there is no such tensor.
     return Tensor(node, port)
 end
 
+Base.getindex(graph::Graph, name) = get_tensor_by_name(graph, name)
+Base.values(graph::Graph) = get_operations(graph)
+Base.keys(graph::Graph) = (node_name(op) for op in values(graph))
+Base.haskey(graph::Graph, name) = isnull(get_node_by_name(graph, name))
+
+# delegate these operations on the Graph for the session that holds it.
+Base.values(sess::Session) = values(sess.graph)
```

Comment:
I see now.
In my own code I use `Session`s when I should in fact just be using the `Graph` at lot of the time, I think.
I basically look at `Session`s as things that hold `Graph`s.
I'll delete it, rather than commenting it out.
Can always dig it out of git history if required.